diff --git a/configure.ac b/configure.ac
index 0a75332..1f465b7 100644
--- a/configure.ac
+++ b/configure.ac
@@ -34,9 +34,7 @@ AC_ARG_ENABLE(guile2,
 
 AC_ARG_WITH(texgyre-dir,
     [AS_HELP_STRING([--with-texgyre-dir=DIR],
-		    [location of TeX Gyre fonts.])],
-    [TEXGYRE_DIR=$withval],
-    [TEXGYRE_DIR=""])
+		    [location of TeX Gyre fonts.])])
 
 reloc_b=no
 AC_ARG_ENABLE(relocation,
@@ -62,47 +60,6 @@ STEPMAKE_COMPILE
 
 AC_CHECK_PROG(FCLIST, fc-list, fc-list)
 AC_MSG_CHECKING([for TeX Gyre fonts OTF files])
-AC_SUBST(TEXGYRE_DIR)
-AC_SUBST(TEXGYRE_FILES)
-if test "$TEXGYRE_DIR" = ""; then
-  if test "$FCLIST" != ""; then
-    TEXGYRE_FILE=`$FCLIST "TeX Gyre Schola:fontformat=CFF" file | head -n 1`
-    if test "$TEXGYRE_FILE" != ""; then
-      TEXGYRE_FILE=`echo $TEXGYRE_FILE | sed 's/\(:.*\)$//g'`
-      TEXGYRE_FILE=`$PYTHON "$srcdir/scripts/auxiliar/readlink.py" $TEXGYRE_FILE`
-      TEXGYRE_DIR=`AS_DIRNAME($TEXGYRE_FILE)`
-    fi
-  fi
-fi
-if test "$TEXGYRE_DIR" != "" ;  then
-  for f in texgyreschola-regular.otf texgyreschola-italic.otf \
-           texgyreschola-bold.otf texgyreschola-bolditalic.otf \
-           texgyreheros-regular.otf texgyreheros-italic.otf \
-           texgyreheros-bold.otf texgyreheros-bolditalic.otf \
-           texgyrecursor-regular.otf texgyrecursor-italic.otf \
-           texgyrecursor-bold.otf texgyrecursor-bolditalic.otf; do
-    if test ! -f "$TEXGYRE_DIR/$f"; then
-      MISSING_TEXGYRE_FILES="$MISSING_TEXGYRE_FILES $f"
-    else
-      TEXGYRE_FILES="$TEXGYRE_FILES $f"
-    fi
-  done
-fi
-if test "$TEXGYRE_DIR" = "" -o "$MISSING_TEXGYRE_FILES" != ""; then
-  AC_MSG_RESULT(no)
-  STEPMAKE_ADD_ENTRY(REQUIRED,TeX Gyre fonts OTF)
-  if test "$TEXGYRE_DIR" = ""; then
-    if test "$FCLIST" = ""; then
-      STEPMAKE_ADD_ENTRY(REQUIRED,'(install the fc-list utility from FontConfig, or use --with-texgyre-dir)')
-    else
-      STEPMAKE_ADD_ENTRY(REQUIRED,"(make sure the fc-list utility can see them, e.g. 'sudo apt-get install fonts-texgyre', or use --with-texgyre-dir)")
-    fi
-  else
-    STEPMAKE_ADD_ENTRY(REQUIRED,'(these files are missing: $MISSING_TEXGYRE_FILES)')
-  fi
-else
-  AC_MSG_RESULT(yes)
-fi
 
 AC_LANG([C++])
 
diff --git a/mf/GNUmakefile b/mf/GNUmakefile
index 4860f65..db22697 100644
--- a/mf/GNUmakefile
+++ b/mf/GNUmakefile
@@ -35,8 +35,6 @@ OTF_TABLES = $(STAFF_SIZES:%=$(outdir)/feta%.otf-table) \
 SVG_FILES = $(OTF_FILES:%.otf=%.svg)
 WOFF_FILES = $(OTF_FILES:%.otf=%.woff)
 
-TEXGYRE_OTFS = $(addprefix $(TEXGYRE_DIR)/,$(TEXGYRE_FILES))
-
 LILYPOND_FONTS_CONF = $(outdir)/00-lilypond-fonts.conf \
 		      $(outdir)/99-lilypond-fonts.conf
 
@@ -86,8 +84,7 @@ INSTALLATION_FILES = $(MF_FILES)
 INSTALLATION_OUT_SUFFIXES = 1 2 3
 
 INSTALLATION_OUT_DIR1 = $(local_lilypond_datadir)/fonts/otf
-INSTALLATION_OUT_FILES1 = $(OTF_FILES) \
-			  $(TEXGYRE_OTFS)
+INSTALLATION_OUT_FILES1 = $(OTF_FILES)
 
 INSTALLATION_OUT_DIR2 = $(local_lilypond_datadir)/fonts/svg
 INSTALLATION_OUT_FILES2 = $(SVG_FILES) $(WOFF_FILES)
